luaui/RmlWidgets/rml_starter/generate-widget.sh

```diff
@@ -0,0 +1,350 @@
+#!/bin/bash
+
+# RML Widget Generator Script
+# Usage: ./generate-widget.sh --name widget_name
+
+# Default values
+WIDGET_NAME=""
+
+# Parse command line arguments
+while [[ $# -gt 0 ]]; do
+    case $1 in
+        --name)
+            WIDGET_NAME="$2"
+            shift 2
+            ;;
+        -h|--help)
+            echo "RML Widget Generator"
+            echo ""
+            echo "Usage: $0 --name widget_name"
+            echo ""
+            echo "Required:"
+            echo "  --name NAME        Widget name (alphanumeric and underscores only)"
+            echo ""
+            echo "Options:"
+            echo "  -h, --help         Show this help message"
+            echo ""
+            echo "Examples:"
+            echo "  $0 --name my_widget"
+            echo "  $0 --name build_menu"
+            echo "  $0 --name unit_stats"
+            echo ""
+            echo "Generated widgets use standard size (300x400) and position (top-left)."
+            echo "Customize size and position in the generated .rcss file as needed."
+            exit 0
+            ;;
+        *)
+            # Support legacy positional argument for backward compatibility
+            if [[ -z "$WIDGET_NAME" ]]; then
+                WIDGET_NAME="$1"
+                shift
+            else
+                echo "Unknown argument: $1"
+                exit 1
+            fi
+            ;;
+    esac
+done
+
+# Validate widget name
+if [[ -z "$WIDGET_NAME" ]]; then
+    echo "Error: Widget name is required!"
+    echo "Usage: $0 --name widget_name"
+    echo "Use --help for more information"
+    exit 1
+fi
+
+# Validate widget name format
+if [[ ! "$WIDGET_NAME" =~ ^[a-zA-Z][a-zA-Z0-9_-]*$ ]]; then
+    echo "Error: Widget name must start with a letter and contain only letters, numbers, underscores, and hyphens"
+    exit 1
+fi
+
+WIDGET_DIR="../${WIDGET_NAME}"
+
+# Check if widget directory already exists
+if [ -d "$WIDGET_DIR" ]; then
+    echo "Error: Widget directory '$WIDGET_DIR' already exists!"
+    exit 1
+fi
+
+echo "Generating RML widget: $WIDGET_NAME"
+echo "Creating directory: $WIDGET_DIR"
+
+# Create widget directory
+mkdir "$WIDGET_DIR"
+
+# Generate the Lua file
+cat > "$WIDGET_DIR/${WIDGET_NAME}.lua" << 'EOF'
+if not RmlUi then
+    return
+end
+
+local widget = widget ---@type Widget
+local utils = VFS.Include("luaui/Include/rml_utilities/utils.lua")
+local ccg = VFS.Include("luaui/Include/rml_utilities/common_class_groups.lua") -- already in model but useful here too for custom class groups
+
+function widget:GetInfo()
+    return {
+        name = "WIDGET_NAME_PLACEHOLDER",
+        desc = "Generated RML widget template",
+        author = "Generated from rml_starter/generate-widget.sh",
+        date = "2025",
+        license = "GNU GPL, v2 or later",
+        layer = -10000,
```

Comment:
is there a specific reason we want this layer to be this negative? Is it possible to use 0?

Reply:
I'd be keen on others weighing in on this, I think you're right and that this choice is a relic of working in the context manager in which we definitely need that widget running early and always.

Reply:
I've decided to keep this in the negative. Interface widgets need to be prepared as early as possible whether they are shown or not  is inconsequential. Layer in this instance applies only to the  order of  the load and not the z-index of the widget.